### PR TITLE
v1.6 backports 2020-02-17

### PR DESCRIPTION
**Pull request by nebril**
* #10113 -- bpf: Fix proxy redirection for egress programs (@tgraf)
 * #10131 -- doc: Document L7 limitation in azure-cni chaining mode (@tgraf)
 * #10144 -- ipcache: Add probe to check for dump capability to support delete (@tgraf)
 * #10140 -- pkg/endpoint: access endpoint state safely across go routines (@aanm)
 * #10137 -- pkg/eventqueue: fix concurrent access of waitConsumeOffQueue (@aanm)
 * #10171 -- charts: Generate versions from VERSION file (@joestringer)
 * #10173 -- bpf: Fix space hack in Makefile (@brb)
 * #10178 -- docs: fix link for Cilium-PR-Kubernetes-Upstream job (@tklauser)
 * #10192 -- maps/nat: return NatEntry6 for NatKey6 NewValue() instead of NatEntry4 (@aanm)
 * #10190 -- bugtool: Dump NAT BPF maps entries with bpftool (@brb)
 * #10185 -- Correct clustermesh identity sync kvstore backend usage (to actually use the remote) (@raybejjani)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 10113 10131 10144 10140 10137 10171 10173 10178 10192 10190 10185; do contrib/backporting/set-labels.py $pr done 1.6; done
```
Note: dropped last commit from #10185, as `status` method was not available in 1.6 branch

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10223)

### CODE REVIEW

**Comment by nebril** [comment]
never-tell-me-the-odds

**Review by aanm — APPROVED** [review]
LGTM for my PRs

**Comment by nebril** [comment]
never-tell-me-the-odds

**Comment by nebril** [comment]
never-tell-me-the-odds

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
never-tell-me-the-odds

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
never-tell-me-the-odds

**Comment by aanm** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
seems like Cilium was not running in runtime vm

**Comment by nebril** [comment]
test-me-please

File: cilium/cmd/kvstore_delete.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ var kvstoreDeleteCmd = &cobra.Command{
 				Fatalf("Unable to delete keys: %s", err)
```
Comment (raybejjani) [inline cilium/cmd/kvstore_delete.go:36]: Not related to this spot, but this backport is missing deleting all the package global functions in `pkg/kvstore/kvstore.go` (https://github.com/cilium/cilium/commit/76c7fcad95a651f22128af96aa8d546682da3cd9#diff-fb5855a8577b97a3f31a2f1fe6c75009L49-L169)

File: pkg/kvstore/allocator/allocator.go [hunk 2]
```diff
@@ -143,15 +146,15 @@ func (k *kvstoreBackend) DeleteAllKeys() {
 	kvstore.Client().DeletePrefix(k.basePrefix)
```
Comment (raybejjani) [inline pkg/kvstore/allocator/allocator.go:146]: Ooops! I forgot to change this in the original PR. I'll make a follow up :/

**Comment by nebril** [comment]
test-me-please

**Review by raybejjani — APPROVED** [review]
LGTM for my clustermesh stuff

**Comment by nebril** [comment]
test-me-please

**Comment by aanm** [comment]
@nebril the tests have been failing for the runtime VM for the same reason, this is probably a regression. I would suggest to drop the PRs that had conflicts, rebase against 1.6, test again and open a new PR with the PRs that had conflicts.

**Comment by nebril** [comment]
@aanm yeah, got really unlucky with provisioning errors on this PR, wanted to make sure this isn't the underlying issue

**Comment by nebril** [comment]
never-tell-me-the-odds

**Comment by nebril** [comment]
Seems like #10168 backport broke the CI, removed it from this PR, will create separate PR with just this backport.

**Comment by joestringer** [comment]
Hit https://github.com/cilium/cilium/issues/10270 , retrying k8s tests.

**Comment by joestringer** [comment]
test-missed-k8s

**Pull request merged**